src/commands/search/youtube.ts

```diff
@@ -0,0 +1,104 @@
+		.addStringOption((builder) =>
+			applyLocalizedBuilder(builder, LanguageKeys.Commands.YouTube.OptionsOrder).addChoices(
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderDate, { value: 'date' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderRating, { value: 'rating' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderRelevance, { value: 'relevance' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderTitle, { value: 'title' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderVideoCount, { value: 'videoCount' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsOrderViewCount, { value: 'viewCount' })
+			)
+		)
+		.addStringOption((builder) =>
+			applyLocalizedBuilder(builder, LanguageKeys.Commands.YouTube.OptionsType).addChoices(
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsTypeChannel, { value: 'channel' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsTypePlaylist, { value: 'playlist' }),
+				createSelectMenuChoiceName(LanguageKeys.Commands.YouTube.OptionsTypeVideo, { value: 'video' })
+			)
+		)
+)
+export class UserCommand extends Command {
+	public override async chatInputRun(interaction: Command.ChatInputInteraction, options: Options) {
+		const result = await this.query(options);
+		const body = result.match({
+			ok: (result) => this.handleOk(interaction, result),
+			err: (error) => this.handleError(interaction, error)
+		});
+		return interaction.reply(body);
+	}
+
+	private query(options: Options) {
+		const url = new URL('https://youtube.googleapis.com/youtube/v3/search');
+		url.searchParams.append('part', 'snippet');
+		url.searchParams.append('safeSearch', 'strict');
+		url.searchParams.append('maxResults', '25');
+		url.searchParams.append('q', options.query);
+		url.searchParams.append('key', envParseString('GOOGLE_API_TOKEN'));
+		if (!isNullishOrEmpty(options.order)) url.searchParams.append('order', options.order);
+		if (!isNullishOrEmpty(options.type)) url.searchParams.append('type', options.type);
+
+		return Json<YouTubeResult>(safeTimedFetch(url, 2000));
+	}
+
+	private handleOk(interaction: Command.ChatInputInteraction, result: YouTubeResult): MessageResponseOptions {
+		let first: YouTubeResultId | null = null;
+		const options: APISelectMenuOption[] = [];
+		for (const item of result.items) {
+			getSelectMenuValue(item.id).inspect((value) => {
+				options.push({
+					// Handle edge case with channelTitle: ''.
+					label: this.getLabel(item.snippet),
+					description: cutText(he.decode(item.snippet.title), 50),
+					value,
+					default: first === null
+				});
+				first ??= item.id;
+			});
+		}
+
+		if (options.length === 0) {
+			const content = resolveUserKey(interaction, LanguageKeys.Commands.YouTube.NoResults);
+			return { content, flags: MessageFlags.Ephemeral };
+		}
```

Comment:
Move this to the top of the function and check for `result.items.length`

Reply:
Not all entries are pushed, only those that returm a `Some<string>` do, as `.inspect` will only be called for those.